polaris-react/src/components/Button/Button.scss

```diff
@@ -554,11 +554,7 @@
     }
 
     &.iconOnly {
-      margin: calc(-1 * var(--p-space-1_5-experimental));
-      @media #{$p-breakpoints-md-up} {
-        margin-left: calc(-1 * var(--p-space-1));
-        margin-right: calc(-1 * var(--p-space-1));
-      }
```

Comment:
This was causing the icon only plain primary button to not be aligned with text elements on mobile.

|Before|After|
|-|-|
|<img width="302" alt="Screenshot 2023-06-21 at 11 35 25 PM" src="https://github.com/Shopify/polaris/assets/20652326/03f31ad7-0484-4c13-9a6d-90676767fa95"> |<img width="305" alt="Screenshot 2023-06-21 at 11 35 58 PM" src="https://github.com/Shopify/polaris/assets/20652326/22b93bcc-45d4-489e-a301-975ecca4e749">|